Px.Utils/Validation/DatabaseValidation/DatabaseValidator.cs

```diff
@@ -0,0 +1,398 @@
+        {
+            if (SyntaxValidator.IsEndOfMetadataSection(character, _syntaxConf, entryBuilder, isProcessingString) && defaultLanguage != string.Empty)
+            {
+                languages = [defaultLanguage.Trim(_syntaxConf.Symbols.Key.StringDelimeter)];
+            }
+            else if (character == _syntaxConf.Symbols.Key.StringDelimeter)
+            {
+                isProcessingString = !isProcessingString;
+            }
+            else if (character == _syntaxConf.Symbols.EntrySeparator && !isProcessingString)
+            {
+                ProcessEntry(entryBuilder, ref defaultLanguage, ref languages);
+                entryBuilder.Clear();
+            }
+            else
+            {
+                entryBuilder.Append(character);
+            }
+        }
+
+        private async Task<DatabaseFileInfo> GetPxFileInfoAsync(string filename, Stream stream, CancellationToken cancellationToken)
+        {
+            string name = _fileSystem.GetFileName(filename);
+            string? path = _fileSystem.GetDirectoryName(filename);
+            string location =  path is not null ? path : string.Empty;
+            string[] languages = [];
+            PxFileMetadataReader metadataReader = new ();
+            Encoding encoding = Encoding.Default;
+            try
+            {
+                encoding = await metadataReader.GetEncodingAsync(stream, _syntaxConf, cancellationToken);
+            }
+            catch (InvalidPxFileMetadataException e)
+            {
+                feedbacks.Add(new(
+                    new(ValidationFeedbackLevel.Error,
+                    ValidationFeedbackRule.NoEncoding),
+                    new(filename, additionalInfo: $"Error while reading the encoding of the file {filename}: {e.Message}"))
+                );
+            }
+            stream.Position = 0;
+            const int bufferSize = 1024;
+            bool isProcessingString = false;
+            using StreamReader streamReader = new(stream, encoding, leaveOpen: true);
+            StringBuilder entryBuilder = new();
+            char[] buffer = new char[bufferSize];
+            string defaultLanguage = string.Empty;
+            int read = 0;
+            do
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                read = await streamReader.ReadAsync(buffer.AsMemory(), cancellationToken);
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ProcessBuffer(buffer[i], ref entryBuilder, ref isProcessingString, ref defaultLanguage, ref languages);
+                }
+            } while (languages.Length == 0 && read > 0);
+
+            DatabaseFileInfo fileInfo = new (name, location, languages, encoding);
+            return fileInfo;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ProcessEntry(StringBuilder entryBuilder, ref string defaultLanguage, ref string[] languages)
```

Comment:
Pass a string as a parameter instead, since no other methods of the entrybuilder are called.